src/Stratis.Bitcoin.Features.BlockStore.Tests/BlockStoreTests.cs

```diff
@@ -0,0 +1,405 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+using Moq;
+using NBitcoin.DataEncoders;
+
+namespace Stratis.Bitcoin.Features.BlockStore.Tests
+{
+    public class BlockStoreTests
+    {
+        private BlockStore blockStore;
+        private readonly IBlockRepository repository;
+        private readonly IChainState chainState;
+        private readonly NodeLifetime nodeLifetime;
+        private ConcurrentChain chain;
+
+        private uint256 repositoryBlockHash;
+        private int repositorySavesCount = 0;
+        private int repositoryTotalBlocksSaved = 0;
+        private int repositoryTotalBlocksDeleted = 0;
+
+        private Dictionary<uint256, Block> listOfSavedBlocks;
+
+        private ChainedHeader chainStateBlockStoreTip;
+
+        private ChainedHeader consensusTip;
+
+        private string testBlockHex = "07000000af72d939050259913e440b23bee62e3b9604129ec8424d265a6ee4916e060000a5a2cbad28617657336403daf202b797bfc4b9c5cfc65a258f32ec33ec9ad485314ea957ffff0f1e812b07000101000000184ea957010000000000000000000000000000000000000000000000000000000000000000ffffffff03510101ffffffff010084d717000000001976a9140099e795d9ee809dc74dce32c79d26db0265072488ac0000000000";
+
+        public BlockStoreTests()
+        {
+            this.listOfSavedBlocks = new Dictionary<uint256, Block>();
+            this.listOfSavedBlocks.Add(uint256.One, Block.Parse(this.testBlockHex, Network.StratisMain));
+
+            this.chain = this.CreateChain(10);
+            this.consensusTip = null;
+            this.nodeLifetime = new NodeLifetime();
+
+            var reposMoq = new Mock<IBlockRepository>();
+            reposMoq.Setup(x => x.PutAsync(It.IsAny<uint256>(), It.IsAny<List<Block>>())).Returns((uint256 nextBlockHash, List<Block> blocks) =>
+            {
+                this.repositoryBlockHash = nextBlockHash;
+                this.repositorySavesCount++;
+                this.repositoryTotalBlocksSaved += blocks.Count;
+                return Task.CompletedTask;
+            });
+
+            reposMoq.Setup(x => x.DeleteAsync(It.IsAny<uint256>(), It.IsAny<List<uint256>>())).Returns((uint256 nextBlockHash, List<uint256> blocks) =>
```

Comment:
I like putting the .Returns on a new line indented in from the line above to make it leap of the page telling you what it returns at a glance.

Reply:
fixed